classes/author/Collector.inc.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @file classes/author/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of authors
+ */
+
+namespace APP\author;
+
+use APP\core\Application;
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+
+class Collector extends \PKP\author\Collector
+{
+    /** @var int|null */
+    public $chapterId = null;
+
+    public function __construct(\PKP\author\DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Limit results to submissions assigned to these series
+     */
+    public function filterByChapter(int $chapterId): self
+    {
+        $this->chapterId = $chapterId;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = parent::getQueryBuilder();
+
+        if ($this->chapterId) {
+            $q->leftJoin('submission_chapter_authors as sca', 'a.author_id', '=', 'sca.author_id');
+            $q->where('sca.chapter_id', '=', $this->chapterId);
+        }
```

Comment:
@asmecher will this work without a `GROUP BY` clause or does he need to use the subquery syntax you recommended?

Reply:
This would need to be a subquery, since the same author might be active on several chapters.

Reply:
Changed